discover/discover.go

```diff
@@ -80,26 +80,23 @@ func ScanAndConnect(host string, username string, password string, options ...Op
 	for _, probeID := range order {
 		probeDevice := devices[probeID]
 
-		opts.Logger.V(1).Info("probing to identify device", "step", "ScanAndConnect", "host", host)
+		opts.Logger.V(1).Info("probing to identify device", "step", "ScanAndConnect", "host", host, "vendor", probeID)
 
-		bmcConnection, err := probeDevice(opts.Context, opts.Logger)
+		bmcConnection, probeErr := probeDevice(opts.Context, opts.Logger)
 
 		// if the device didn't match continue to probe
-		if err != nil && (err == errors.ErrDeviceNotMatched) {
+		if probeErr != nil {
+			// log error if probe is not successful
+			opts.Logger.V(1).Info("probe failed", "host", host, "vendor", probeID, "error", probeErr)
```

Comment:
Should we include the step as well to keep the same structure as the previous log message?

Reply:
sure, will update